atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/ValidatingTransactionScopedCache.java

```diff
@@ -180,7 +180,7 @@ private void validateCacheRowReads(
             TableReference tableReference,
             NavigableMap<byte[], RowResult<byte[]>> remoteReads,
             NavigableMap<byte[], RowResult<byte[]>> cacheReads) {
-        if (!ByteArrayUtilities.areRowResultsEqual(remoteReads, cacheReads)) {
+        if (!remoteReads.equals(cacheReads)) {
```

Comment:
discussed offline: I think I misspoke in saying they were necessarily equivalent 🔥

Reply:
actually, I misspoke that I misspoke in saying they were necessarily equivalent :(

Reply:
this works as long as you can get from the other map, which you can.

Reply:
another U-turn: I misspoke that I misspoke that I misspoke in saying they were necessarily equivalent.

This is correct in terms of the implementation of `TreeMap`. However, the contract of `equals()` is that 

```
... two maps {@code m1} and
     * {@code m2} represent the same mappings if
     * {@code m1.entrySet().equals(m2.entrySet())}.
```

We would be relying on an implementation detail of the tree-map.